includes/class-newspack-popups-segmentation.php

```diff
@@ -181,5 +184,36 @@ function( $cat ) {
 	public static function get_segmentation_endpoint() {
 		return plugins_url( '../api/segmentation/index.php', __FILE__ );
 	}
+
+	/**
+	 * Create the clients and events tables.
+	 */
+	public static function create_database_table() {
+		global $wpdb;
+		$events_table_name = Segmentation::get_events_table_name();
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		if ( $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $events_table_name ) ) != $events_table_name ) {
+			$charset_collate = $wpdb->get_charset_collate();
+
+			$sql = "CREATE TABLE $events_table_name (
```

Comment:
The table structure should include indexes for all the fields that will appear in `WHERE` clauses, and if we do decide to enforce one row per `client_id`/`post_id` combo we'll need a multi-column `UNIQUE` key. Some good examples can be found in the `CREATE` statements for WP taxonomy tables: https://github.com/WordPress/WordPress/blob/d5b8d282e8c03929aafb1fa660181b7db9ad55d3/wp-admin/includes/schema.php#L65-L91

Reply:
Done in b1c6e4a645776590017cb7e684dbc43db1b151fd, along with `INSERT ON DUPLICATE KEY UPDATE`